include/swift/AST/DiagnosticsSema.def

```diff
@@ -910,18 +910,22 @@ ERROR(super_initializer_not_in_initializer,none,
 WARNING(isa_is_always_true,none, "'%0' test is always true",
         (StringRef))
 WARNING(isa_is_foreign_check,none,
-      "'is' test is always true because %0 is a Core Foundation type",
-      (Type))
+        "'is' test is always true because %0 is a Core Foundation type",
+        (Type))
 WARNING(conditional_downcast_coercion,none,
-      "conditional cast from %0 to %1 always succeeds",
-      (Type, Type))
-
+        "conditional cast from %0 to %1 always succeeds",
+        (Type, Type))
+WARNING(unecessary_same_type_coercion,none,
+        "casting expression to %0 doesn't change the type",
+        (Type))
+WARNING(unecessary_same_typealias_type_coercion,none,
+        "casting expression of type %0 to %1 doesn't change the type",
```

Comment:
Maybe `redundant cast from type %0 to %1`?

Reply:
Humm 🤔 the message here, was just the one mentioned in the SR by @harlanhaskins 
I like the last one, seems more simple, but I think if we change the second we should probably change the first one to something like `"redundant cast to %0"` instead of `"casting expression to %0 doesn't change the type"`

Reply:
`redundant cast from %0 to %0 has no effect`, or even reuse the wording of `conditional_downcast_coercion` and say `cast from %0 to %1 always succeeds`.

Reply:
@CodaFi The first one seems nice. I actually prefer `has no effect` instead of reusing `always succeeds` :)